extensions/amp-stream-gallery/0.1/amp-stream-gallery.js

```diff
@@ -0,0 +1,752 @@
+   */
+  updateMaxVisibleCount_(maxVisibleCount) {
+    this.maxVisibleCount_ = maxVisibleCount || Number.MAX_VALUE;
+    this.updateVisibleCount_();
+  }
+
+  /**
+   *
+   * @param {number} minItemWidth
+   */
+  updateMinItemWidth_(minItemWidth) {
+    this.minItemWidth_ = minItemWidth || 1;
+    this.updateVisibleCount_();
+  }
+
+  /**
+   *
+   * @param {number} minVisibleCount
+   */
+  updateMinVisibleCount_(minVisibleCount) {
+    this.minVisibleCount_ = minVisibleCount || 1;
+    this.updateVisibleCount_();
+  }
+
+  /**
+   * Updates the number of items visible for the internal carousel based on
+   * the min/max item widths and how much space is available.
+   */
+  updateVisibleCount_() {
+    const {
+      maxItemWidth_,
+      minItemWidth_,
+      maxVisibleCount_,
+      minVisibleCount_,
+      slides_,
+    } = this;
+    // Need to subtract out the width of the next/prev arrows. If these are
+    // inset, they will have no width.
+    const width =
+      this.element./* OK */ getBoundingClientRect().width -
+      this.getWidthTakenByArrows_();
+    const items = this.getItemsForWidth_(width, minItemWidth_);
+    const maxVisibleSlides = Math.min(slides_.length, maxVisibleCount_);
+    // Cannot use clamp, maxVisibleSlides can be less than minVisibleCount_.
+    const visibleCount = Math.min(
+      Math.max(minVisibleCount_, items),
+      maxVisibleSlides
+    );
+    const advanceCount = Math.floor(visibleCount);
+
+    this.mutateElement(() => {
+      /*
+       * When we are going to show more slides than we have, cap the width so
+       * that we do not go over the max requested slide width. Otherwise,
+       * cap the max width based on how many items are showing and the max
+       * width for each item.
+       */
+      const maxContainerWidth =
+        items > maxVisibleSlides
+          ? `${maxVisibleSlides * maxItemWidth_}px`
+          : `${items * maxItemWidth_}px`;
+
+      setStyle(this.slidesContainer_, 'max-width', maxContainerWidth);
+    });
+    this.carousel_.updateSlides(this.slides_);
+    this.carousel_.updateAdvanceCount(advanceCount);
+    this.carousel_.updateSnapBy(advanceCount);
+    this.carousel_.updateVisibleCount(visibleCount);
+  }
+
+  /**
+   * @param {boolean} outsetArrows
+   * @private
+   */
+  updateOutsetArrows_(outsetArrows) {
+    this.outsetArrows_ = outsetArrows;
+    this.updateUi_();
+  }
+
+  /**
+   * @param {string} extraSpace
+   * @private
+   */
+  updateExtraSpace_(extraSpace) {
+    this.content_.setAttribute(
+      'i-amphtml-stream-gallery-extra-space',
+      extraSpace
+    );
+  }
+
+  /**
+   * @param {string} insetArrowVisibility
+   * @private
+   */
+  updateInsetArrowVisibility_(insetArrowVisibility) {
+    this.insetArrowVisibility_ =
+      insetArrowVisibility == 'always'
+        ? InsetArrowVisibility.ALWAYS
+        : insetArrowVisibility == 'never'
+        ? InsetArrowVisibility.NEVER
+        : InsetArrowVisibility.AUTO;
```

Comment:
This is a bit tricky to read; might be worth considering either adding parens or

```
this.insetArrowVisibility = {
  always: InsetArrowVisibility.ALWAYS
  never: InsetArrowVisibility.NEVER
}[insetArrowVisibility] || InsetArrowVisibility.AUTO;
```

Reply:
I changed this (pulled the `Object` up to the top), but I think it needs to use a `dict()` to prevent renaming of the fields.

Reply:
:+1: